find.go

```diff
@@ -475,16 +475,27 @@ func Nth[T any, N constraints.Integer](collection []T, nth N) (T, error) {
 
 // Sample returns a random item from collection.
 func Sample[T any](collection []T) T {
+	result := SampleBy(collection, rand.IntN)
+	return result
+}
+
+// SampleBy returns a random item from collection, using intFunc as the random index generator.
+func SampleBy[T any](collection []T, intFunc func(int) int) T {
```

Comment:
Why this method doesn't call SamplesBy with count=1 and return the first entry?

Reply:
Also the original implementation of the `Sample` doesn't use `Samples`. I didn't want to change any implementation of existing functions. Also... not sure if there's that much of a benefit in doing so?

Reply:
But if you want me to update it... by all means and I change it!

Reply:
Let's wait for owner's feedback

cc @samber